src/vse_sync_pp/analyzers/analyzer.py

```diff
@@ -137,3 +140,53 @@ def test(self, data):
     def explain(self, data):
         """Return a structured analysis of the collected `data`"""
         raise NotImplementedError
+
+class TimeErrorAnalyzerBase(Analyzer):
+    """Analyze time error"""
+    def __init__(self, config):
+        super().__init__(config)
+        # required system time output accuracy
+        accuracy = config.requirement('time-error-in-locked-mode/ns')
+        # limit on inaccuracy at observation point
+        limit = config.parameter('time-error-limit/%')
+        # exclusive upper bound on absolute time error for any sample
+        self._unacceptable = accuracy * (limit / 100)
+        # samples in the initial transient period are ignored
+        self._transient = config.parameter('transient-period/s')
+        # minimum test duration for a valid test
+        self._duration = config.parameter('min-test-duration/s')
+        # default locked value
+        self._lockid = frozenset({'s2'})
```

Comment:
Move into ts2phc analyzer?

Reply:
Makes sense. `lockid` has passed to class attribute declared only in the child classes.